### ISSUE TITLE
[TorchScript/JIT] del statements with multiple targets not correctly handled

### ISSUE THREAD

**Issue by SplitInfinity** [pytorch/pytorch#37617]
## 🐛 Bug

The Python reference manual states that `del` statements can contain multiple targets (i.e. `del a, b, c` is valid). This works in Python, but does not work in TorchScript.

## To Reproduce
Run the following function in Python and TorchScript:
```
 def fn(a):
     # type: (List[int]) -> List[int]
     del a[0], a[1]
     return a
```

## Expected behavior
The expected behaviour is that the function returns the same list it was given, but with the first two elements removed, just as in Python:
```
 def fn(a):
     # type: (List[int]) -> List[int]
     del a[0], a[0]
     return a
 
 
 if __name__ == "__main__":
     print(fn([1, 2, 3, 4]))

$ python3 test.py
[3, 4]
```

## Actual behaviour
```
@torch.jit.script
 def fn(a):
     # type: (List[int]) -> List[int]
     del a[0], a[0]
     return a
 
 
 if __name__ == "__main__":
     print(fn([1, 2, 3, 4]))

$ python3 test.py
[2, 3, 4]
```

## Environment
```
$ python3 collect_env.py
PyTorch version: 1.6.0a0+3769ca3
Is debug build: No
CUDA used to build PyTorch: None

OS: Mac OSX 10.14.6
GCC version: Could not collect
CMake version: version 3.14.0

Python version: 3.7
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip3] numpy==1.17.4
[conda] blas                      1.0                         mkl  
[conda] mkl                       2019.4                      233  
[conda] mkl-include               2020.0                      166  
[conda] mkl-service               2.3.0            py37hfbe908c_0  
[conda] mkl_fft                   1.0.14           py37h5e564d8_0  
[conda] mkl_random                1.1.0            py37ha771720_0  
[conda] numpy                     1.17.2           py37h99e6662_0  
[conda] numpy-base                1.17.2           py37h6575580_0  
[conda] numpydoc                  0.9.1                      py_0  
[conda] torch                     1.6.0a0+3769ca3           dev_0    <develop>
```

cc @suo

**Comment by eellison** [comment]
Just curious - is this something you ran into in user code ?

**Comment by SplitInfinity** [comment]
No, I was just messing around while working on #37608. Based on my reading of the parser code, I thought `del a, b, c` would be a parsing error (which is better than producing incorrect results) but I tried it out and then this happened.

**Comment by eellison** [comment]
Word, nice catch. Yea if it's silently wrong semantics we should definitely fix it. In general i'm wary of expanding features without a user request because increase our surface area without much benefit; `del` is an example of this. But definitely incorrect compilation as opposed to hard errors should be fixed.

**Comment by SplitInfinity** [comment]
Alright, I did some investigation and it looks like the problem is this piece of the Python frontend:
```
    @staticmethod
    def build_Delete(ctx, stmt):
        return Delete(build_expr(ctx, stmt.targets[0]))
```

I think the easiest thing to do here is to raise a `NotSupportedError` if there is more than one target for the `Delete` statement. This matches the implicit expectation in the C++ frontend that there is only one target:
```
      case TK_DELETE: {
        L.expect(TK_DELETE);
        auto expr = parseExp();
        L.expect(TK_NEWLINE);
        return Delete::create(expr);
      }
```
A statement like `del a[0], b[0]` results in a parsing error because there is a comma where the parser expects a newline.

**Comment by SplitInfinity** [comment]
This was fixed in #38089
